### ISSUE TITLE
Pod Delete Experiment does not completely shut down after termination invoked by chaosCurrentTimeStamp hard stop

### ISSUE THREAD

**Issue by cpitstick-argo** [litmuschaos/litmus#1511]
**What happened**:
On EKS v1.15 and v1.16, ran the `pod-delete` experiment with `litmuschaos/ansible-runner:latest` with the following settings:
```
TOTAL_CHAOS_DURATION: 3600
CHAOS_INTERVAL: 1
```

And while, due to https://github.com/litmuschaos/litmus/pull/1492, chaos does stop, the actual experiment doesn't terminate. The `pod-delete` pod is still active long after the experiment halts, and no events indicating the conclusion of the experiment are emitted:

When I get the logs from the `pod-delete` pod, they look like this:

```
2020-05-19 20:42:16.496737 Step: [Status]: Checking whether application containers are in running state 
task path: /utils/common/status_app_pod.yml:9
details => {"attempts": 3, "changed": true, "cmd": "kubectl get pod -n hello-world-gamma -l app=hello-world-gamma-application --no-headers -o jsonpath='{.items[*].status.containerStatuses[*].ready}' | tr ' ' '\\n' | uniq", "delta": "0:00:00.117135", "end": "2020-05-19 20:42:16.419769", "rc": 0, "start": "2020-05-19 20:42:16.302634", "stderr": "", "stderr_lines": [], "stdout": "true", "stdout_lines": ["true"]}

2020-05-19 20:42:16.705456 Step: [Prepare]: Getting the chaos current timestamp 
task path: /chaoslib/litmus/pod_delete/kill_random_pod.yml:4
details => {"ansible_facts": {"chaosCurrentTimeStamp": "2020-05-19 20:42:16"}, "changed": false}

2020-05-19 20:42:16.792701 Step: set_fact 
task path: /chaoslib/litmus/pod_delete/kill_random_pod.yml:9
details => {"ansible_facts": {"chaosDiffTimeStamp": "3610.0"}, "changed": false}

2020-05-19 20:42:17.969952 Step: [Mark]: Setting chaos_status to terminate 
task path: /chaoslib/litmus/pod_delete/kill_random_pod.yml:114
details => {"ansible_facts": {"chaos_status": "terminate"}, "changed": false}
```

NOTE: The above is the end of the log, there is nothing more. ksatchit@ originally asked me to provide "the rest of it." That's it, there is no more, and that's the problem.

**What you expected to happen**:
Experiment runs for an hour, stops injecting chaos, and then terminates pods and shuts down. Events such as `ChaosEngineCompleted` and `PostChaosCheck` are emitted.

**Comment by cpitstick-argo** [comment]
Here's an example of what happens:

![Screen Shot 2020-05-19 at 10 21 31 PM](https://user-images.githubusercontent.com/57147871/82480173-6863bb00-9aa1-11ea-8460-81d44f33ab78.png)

**Comment by ksatchit** [comment]
This seems to be caused due to the playbook execution slowing down drastically as the iterations progress -- which in turn happens due to the progressive increase in amount of memory consumed by the run. In the ansible-based pod-delete experiment, this is alleviated by moving to the "helper pod approach" (ref: https://github.com/litmuschaos/litmus/pull/1510)  wherein the kills occur via a non-ansible based injection. The memory consumed with this approach is much lesser and is capped to a relatively smaller upper limit - thereby achieving both the hardstop (+ few sec toleration) and iteration count in a predictable way.  At this point, 
these are the resource util values for a successful 3600s/1s run: 

[Report.txt](https://github.com/litmuschaos/litmus/files/4679995/Report.txt)

Currently, event generation from the helper pods is being worked on. 

In case of the go-based pod-delete experiment (which uses less resources), support has been added for force deletes with less gracePeriod to ensure increased iterations of chaos within the prescribed duration. Here, the pending activity/issue is the ongoing investigation of suppressed/clustered events that are being seen when continuous runs (1s interval) occur.

**Comment by ksatchit** [comment]
cc: @uditgaurav @ispeakc0de -- who are championing this !!

**Comment by ksatchit** [comment]
This was fixed via changes in 1.4.1 & 1.5.0: 

- Usage of helper pods reduced the memory consumption on the ansible experiment
- Usage of direct api call to create events over usage of client-go/recorder helped with consistent event generation
- A time diff of currentChaosTimestamp and TotalDuration is used to a achieve a hard-stop over the older iteration based approach
